File: 2023-08-04/e/main.cpp
#include <iostream>
#include <string>

//
// https://www.geeksforgeeks.org/prime-numbers/
//
bool is_prime(int n, int i)
{
  if( n == 0 || n == 1 )
    return false;

  if( n == i )
    return true;

  if( n % i == 0)
    return false;

  i += 1;

  return is_prime(n,i);

}

int main() {
    int n; 
    std::cin >> n;

    for( int m = 1; m < 1001; m++)
    {
      int p = n*m + 1;
      if( ! is_prime( p, 2 ) )
      {
          std::cout << m << std::endl;
          break;
      }
    }
    return 0;
}

File: 2023-08-04/e/main.py
    #
# https://www.geeksforgeeks.org/prime-numbers/
#
def _is_prime(n , i):
    if (n == 0 or n == 1):
        return False

    if (n == i):
        return True

    if (n % i == 0):
        return False

    i += 1

    return _is_prime(n,i)
    

def is_prime(n):
    return _is_prime(n, 2)


def main():

    n = int( input() )
    
    for m in range(1,1001):
        if not is_prime(n*m + 1):
            return m


if __name__ == '__main__':
    print( main() )
